### ISSUE TITLE
Slow Sensor Updates

### ISSUE THREAD

**Issue by segfaultex** [briis/unifiprotect#452]
### Home Assistant Version?

2021.12.9

### UniFi Protect Version?

1.20.3

### UniFi Protect HA Integration version?

0.12.0-beta8

### What happened?

This has actually been happening for a while now.
Pretty much ever since the integration moved away from being loaded from a config file.

It seems like updates from all the sensors are delayed 15 or so seconds.

I've had an automation that triggers when it detects a person/motion from my doorbell from the beginning.
It used to be almost instant.
Now it is very delayed.

I get an email from unifi with the detection before the sensors update and trigger the automation.

Now a person can actually get to the door and ring the doorbell before the motion/person detection automation is triggered.

**Comment by AngellusMortis** [comment]
This is likely not an issue we can solve. It is most likely one of:

* the CPU for your HA instance is getting overloaded and the Websocket messages are not being processed in a timely manner
* the CPU of your UniFi Protect instance is being overloaded and events are getting delayed

Some thoughts on troubleshooting:

* Open the UniFi Protect Web app and HA at the same time. Compare them and see there is a delay between the two. They should both update instantly.
* Check out the [profiler integration](https://www.home-assistant.io/integrations/profiler/) (if you make a profile post it here
* Make sure you **do not** have the "Real-time Metrics" config option enabled (sounds like this has been happening a while though so doubt it is that)

**Comment by AngellusMortis** [comment]
Out of curiosity, what console do you have and how many/what cameras do you have?

**Comment by segfaultex** [comment]
> This is likely not an issue we can solve with a code fix. It is most likely one of:
> 
> * the CPU for your HA instance is getting overloaded and the Websocket messages are not being processed in a timely manner

My HA is running in a container averaging .1 of a cpu core usage so i'm sure it isn't that.

> * the CPU of your UniFi Protect instance is being overloaded and events are getting delayed

The Protect instance is running on a UDM Pro. Since i get email notifications before the sensor changes i really don't think it is on the Unifi end.

> Some thoughts on troubleshooting to help you track down the cause:
> 
> * Open the UniFi Protect Web app and HA at the same time. Compare them and see there is a delay between the two. They should both update instantly. If they are both happening at the same time in UniFi Protect and HA, there is nothing we can do to help you, it is your Protect instance having issues. Post on the UniFi Community Forums.

I tried this and Protect updates instantly but the HA sensors lag behind.

> * Check out the [profiler integration](https://www.home-assistant.io/integrations/profiler/) (if you make a profile post it here

I'll look at the profiler

> * Make sure you **do not** have the "Real-time Metrics" config option enabled (sounds like this has been happening a while though so doubt it is that)

I made sure this is not enabled.


> Out of curiosity, what console do you have and how many/what cameras do you have?

It is running on the UDM Pro with 7 cameras.

**Comment by AngellusMortis** [comment]
It sounds like your HA instance is lagging behind then (CPU on HA). 

> My HA is running in a container averaging .1 of a cpu core usage so i'm sure it isn't that.

That is normal. The main HA progress is only single core.

**Comment by segfaultex** [comment]
> That is normal. The main HA process is only single core. Is it constantly at 100% CPU (one full core)?

The hardware is not the issue.
![image](https://user-images.githubusercontent.com/3958108/149042707-e7389192-f94d-4383-bb7d-e00b8d1a029e.png)

**Comment by AngellusMortis** [comment]
Oh sorry, I missed you said _0_.1 CPU. I did not notice the . at first. 

Then we would definitely need some [debug logs](https://github.com/briis/unifiprotect#enable-debug-logging) and a [profile](https://www.home-assistant.io/integrations/profiler/) to try to help find any more issues.

I also noticed that your `top` output is very limited so it looks you like you ran it inside of Docker container. So, make sure you host system is not using up a lot of CPU as well or in some other way throttling your HA instance. Are you using Home Assistant OS?

**Comment by segfaultex** [comment]
No worries.
It is actually running the official docker image in kubernetes with socat in injetcted for access to the zigbee/zwave dongle.

I think I found the culprit and it isnt the integration.
When I was looking at the sensor changes apparently my phone was swapping between wifi and lte since I was outside. So it looked delayed but it was just a delay to the phone app.

The actual culprit to the automation seeming to be delayed was due to the "turn on" command being disabled on chromecast devices without a screen at some point in an HA update, which the automation was using.

Thanks for helping me troubleshoot this even though it ended up not being anything to do with the integration.
